crates/sui-graphql-rpc/src/context_data/db_data_provider.rs

```
@@ -129,25 +149,222 @@ impl PgManager {
+                .filter(transactions::dsl::tx_sequence_number.lt(before))
+                .order(transactions::dsl::tx_sequence_number.desc());
+        }
+
+        if let Some(filter) = filter {
+            // Filters for transaction table
+            if let Some(kind) = filter.kind {
+                query = query.filter(transactions::dsl::transaction_kind.eq(kind as i16));
+            }
+            if let Some(checkpoint) = filter.checkpoint {
+                query = query
+                    .filter(transactions::dsl::checkpoint_sequence_number.eq(checkpoint as i64));
+            }
+
+            // Filters for tx_indices table
+            match (filter.package, filter.module, filter.function) {
+                (Some(p), None, None) => {
+                    query = query.filter(
+                        tx_indices::dsl::packages.contains(vec![Some(p.into_array().to_vec())]),
+                    );
+                }
+                (Some(p), Some(m), None) => {
+                    query = query.filter(
+                        tx_indices::dsl::package_modules
+                            .contains(vec![Some(format!("{}::{}", p, m))]),
+                    );
+                }
+                (Some(p), Some(m), Some(f)) => {
+                    query = query.filter(
+                        tx_indices::dsl::package_module_functions
+                            .contains(vec![Some(format!("{}::{}::{}", p, m, f))]),
+                    );
+                }
+                _ => {}
+            }
+            if let Some(sender) = filter.sent_address {
+                query = query.filter(tx_indices::dsl::senders.contains(vec![sender.into_vec()]));
+            }
+            if let Some(receiver) = filter.recv_address {
+                query =
+                    query.filter(tx_indices::dsl::recipients.contains(vec![receiver.into_vec()]));
+            }
+            // TODO: sign_, paid_address, input_, changed_object
```

If this isnt implemented yet, let it fail explicitly